tests/Models/MessageModelTest.php

```diff
@@ -0,0 +1,231 @@
+        // Check sent messages again, now there should be 3 entries.
+        $messagesSent = $em->getRepository(Character::class)->find(2)->listSentMessages();
+        $found = 0;
+        foreach ($messagesSent as $message) {
+            if ($message->getAuthor()->getId() === 2
+                && $message->getTitle() === "Hello testSendAndReceiving"
+            ) {
+                $found++;
+            }
+        }
+        $this->assertSame(3, $found);
+        
+        // Check received messages again, now there should be 3 entries.
+        $messagesReceived = $em->getRepository(Character::class)->find(1)->listReceivedMessages();
+        $found = 0;
+        foreach ($messagesReceived as $message) {
+            if ($message->getAddressee()->getId() === 1
+                && $message->getTitle() === "Hello testSendAndReceiving"
+            ) {
+                $found++;
+            }
+        }
+        $this->assertSame(3, $found);
+    }
+    
+    /**
+     * @expectedException \LotGD\Core\Exceptions\InvalidModelException
+     * @expectedExceptionMessage The Addressee of Message model must not be null.
+     */
+    public function testEmptyAddresseeException()
+    {
+        $newMessage = Message::create([
+            "title" => "Hello",
+            "body" => "Hello World.",
+        ]);
+        
+        $newMessage->save($this->getEntityManager());
+    }
+    
+    /**
+     * @expectedException \LotGD\Core\Exceptions\InvalidModelException
+     * @expectedExceptionMessage Author of Message model cannot be empty without beeing a system message.
+     */
+    public function testEmptyAuthorException()
+    {
+        
+        $em = $this->getEntityManager();
+        
+        $newMessage = Message::create([
+            "title" => "Hello",
+            "body" => "Hello World.",
+        ]);
+        
+        $newMessage->setAddressee($em->getRepository(Character::class)->find(1));
+        
+        $newMessage->save($em);
+    }
+    
+    public function testTimezone()
+    {
+        $em = $this->getEntityManager();
+        
+        $time1 = $em->getRepository(Message::class)->find(1)->getCreationTime();
+        $time2 = new \DateTime("2016-05-03 14:19:12");
+        $time3 = $time2->setTimezone(new \DateTimeZone("Europe/Zurich"));
+        $time4 = new \DateTime("2016-05-03 14:19:12");
+        $time4->setTimezone(new \DateTimeZone("America/Los_Angeles"));
+        
+        $this->assertSame($time1->getTimestamp(), $time2->getTimestamp());
+        $this->assertEquals($time1, $time2);
+        $this->assertSame($time2, $time3);
+        $this->assertEquals($time2->getTimezone(), $time3->getTimezone());
+        $this->assertNotEquals($time1->getTimezone(), $time2->getTimezone());
+        $this->assertSame($time1->getTimestamp(), $time3->getTimestamp());
+        $this->assertNotEquals($time2->getTimezone(), $time4->getTimezone());
+    }
+    
+    /*public function dataCreateSaveAndRetrieve(): array
```

Comment:
Let's not check in commented code. It's fine to check something in then remove it immediately in another commit if you really want to store it, but commented out code has a bad habit of just laying around and by the time it gets uncommented, it wont run anymore.